src/personaclient/PersonaClient.php

```diff
@@ -255,11 +269,15 @@ public function isPresignedUrlValid($url,$secret) {
      */
     protected function cacheToken($cacheKey,$token,$expiryTime) {
         // cache this freshly obtained token so we don't have to round-trip to persona again
-        $this->getCacheClient()->transaction(function($tx) use ($cacheKey, $token, $expiryTime) {
-            $tx->multi();
-            $tx->set($cacheKey,json_encode($token));
-            $tx->expire($cacheKey,$expiryTime); // cache for token expiry minus 60s
-        });
+        $cacheClient = $this->getCacheClient();
+        if($cacheClient)
+        {
+            $cacheClient->transaction(function($tx) use ($cacheKey, $token, $expiryTime) {
+                $tx->multi();
+                $tx->set($cacheKey,json_encode($token));
+                $tx->expire($cacheKey,$expiryTime); // cache for token expiry minus 60s
```

Comment:
thats not `expiry minus 60s`, it looks to me like this is always caching the token for 60s